[clang-analyze] Fix assorted clang-analyze warnings

fuzzard opened the pull request:
## Description
Fix various warnings from a clang-analyze report on macos build.

## Motivation and context
Fix some warnings from clang-analyze. This is only some of the low hanging warnings

## How has this been tested?
build osx

## What is the effect on users?
N/A

## Screenshots (if appropriate):

## Types of change
<!--- What type of change does your code introduce? Put an `x` in all the boxes that apply like this: [X] -->
- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [x] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:
<!--- Go over all the following points, and put an `X` in all the boxes that apply like this: [X] -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

Commit:
[clang-analyze] default init fixes

clang analyze picks a number of variables without default init. for example the following

xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoPPFFmpeg.cpp:104:73: warning: The left operand of '*' is a garbage value
                                                          srcStrides[0] * pPicture->iHeight +
                                                          ~~~~~~~~~~~~~ ^

Commit:
[clang-analyze] fix warnings for DVDAudioCodecPassthrough

/Users/brent/macos/xbmc/cores/VideoPlayer/DVDCodecs/Audio/DVDAudioCodecPassthrough.cpp:149:7: warning: Value stored to 'used' is never read
      used = iSize;
      ^      ~~~~~
/Users/brent/macos/xbmc/cores/VideoPlayer/DVDCodecs/Audio/DVDAudioCodecPassthrough.cpp:161:5: warning: Value stored to 'used' is never read
    used = iSize;
    ^      ~~~~~
2 warnings generated.

Commit:
[clang-analyze] fix never read var usage

xbmc/cores/VideoPlayer/VideoRenderers/LinuxRendererGL.cpp:1768:5: warning: Value stored to 'ret' is never read
    ret = false;
    ^     ~~~~~
1 warning generated.

xbmc/filesystem/NFSFile.cpp:85:3: warning: Value stored to 'tmp' is never read
  tmp = exportlist;
  ^     ~~~~~~~~~~
1 warning generated.

lrusak commented on xbmc/cores/VideoPlayer/DVDCodecs/Audio/DVDAudioCodecFFmpeg.cpp:
```diff
@@ -184,7 +184,7 @@ void CDVDAudioCodecFFmpeg::GetData(DVDAudioFrame &frame)
   frame.nb_frames = 0;
 
-  uint8_t* data[16];
+  uint8_t* data[16]{};
```
This seems odd as it's used by the next line.

garbear replied:
You could use `std::array`

fuzzard replied:
RE std:array, not really looking to change behaviour at this point.
Same reasoning as the other one, a couple brackets removes a static analysis warning.

lrusak commented on xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoPPFFmpeg.cpp:
```diff
@@ -96,7 +96,7 @@ void CDVDVideoPPFFmpeg::Process(VideoPicture* pPicture)
 
   uint8_t* srcPlanes[YuvImage::MAX_PLANES], *dstPlanes[YuvImage::MAX_PLANES];
-  int srcStrides[YuvImage::MAX_PLANES];
+  int srcStrides[YuvImage::MAX_PLANES]{};
```
Same thing here. Seems odd but I guess default init doesn't hurt.

fuzzard replied:
for the sake of some brackets to remove a warning from a static analysis tool, id just put it in personally.

lrusak commented on xbmc/guilib/GUIFontTTFGL.cpp:
```diff
@@ -139,18 +139,6 @@ void CGUIFontTTFGL::LastEnd()
 
     // Deal with vertices that had to use software clipping
     std::vector<SVertex> vecVertices(6 * (m_vertex.size() / 4));
-    SVertex *vertices = &vecVertices[0];
```
This seems wrong. This block adds the vertices needed to vecVertives which is used in the gl calls below. There is some pointer addition happening which may be why it thinks the variable is unused?

fuzzard replied:
Yeah, theres some pointer arithmetic, but in the end vertices isnt used for anything in this HAS_GL codepath.
In the GLES equivalent, its used in the following https://github.com/xbmc/xbmc/blob/7823116bdc46bf7668b57585669a107421ff39d2/xbmc/guilib/GUIFontTTFGL.cpp#L199-L201

Whereas the GL path does https://github.com/xbmc/xbmc/blob/7823116bdc46bf7668b57585669a107421ff39d2/xbmc/guilib/GUIFontTTFGL.cpp#L149-L154

lrusak replied:
The gl method uses a GL_ARRAY_BUFFER

https://github.com/xbmc/xbmc/blob/7823116bdc46bf7668b57585669a107421ff39d2/xbmc/guilib/GUIFontTTFGL.cpp#L147

fuzzard replied:
ah, ok, i see. pointer to the pointer to the reference. Ill drop this then. drum it up to a false positive

Commit:
[clang-analyzer] fix warning uninit value

xbmc/cores/AudioEngine/Sinks/darwin/CoreAudioHelpers.cpp:19:29: warning: 1st function call argument is an uninitialized value
  if (isprint(buffer[1]) && isprint(buffer[2]) &&
                            ^~~~~~~~~~~~~~~~~~
1 warning generated.

Commit:
[clang-analyzer] fix warning uninit value

tmpPos has no default init. If the call to dll_fgetpos64 returns EINVAL we dont check
that return value, and just assign to pos.
Im guessing we should be checking the retvalue, but i dont have a test case to verify.

xbmc/cores/DllLoader/exports/emu_msvcrt.cpp:1487:10: warning: Assigned value is garbage or undefined
    *pos = (fpos_t)tmpPos;
         ^ ~~~~~~~~~~~~~~

Call path for warning
xbmc/cores/DllLoader/exports/emu_msvcrt.cpp:1482:5: 'tmpPos' declared without an initial value
xbmc/cores/DllLoader/exports/emu_msvcrt.cpp:1485:11: Calling 'dll_fgetpos64'
xbmc/cores/DllLoader/exports/emu_msvcrt.cpp:1494:3: Entered call from 'dll_fgetpos'
xbmc/cores/DllLoader/exports/emu_msvcrt.cpp:1497:9: Assuming 'pFile' is equal to NULL
xbmc/cores/DllLoader/exports/emu_msvcrt.cpp:1507:5: Returning without writing to '*pos'
xbmc/cores/DllLoader/exports/emu_msvcrt.cpp:1485:11: Returning from 'dll_fgetpos64'
xbmc/cores/DllLoader/exports/emu_msvcrt.cpp:1487:10: Assigned value is garbage or undefined

Commit:
[comment] Add Todo comment regarding unused var in RPBaseRenderer

Garbear believes this may be fallout from bringing in the display integer scaling.
Has requested to add a todo comment for later investigation

kambala-decapitator commented on xbmc/cores/VideoPlayer/DVDCodecs/Audio/DVDAudioCodecPassthrough.cpp:
```diff
@@ -134,7 +133,7 @@ bool CDVDAudioCodecPassthrough::AddData(const DemuxPacket &packet)
 
     m_dataSize = m_bufferSize;
-    used = m_parser.AddData(pData, iSize, &m_buffer, &m_dataSize);
+    int used = m_parser.AddData(pData, iSize, &m_buffer, &m_dataSize);
```
```suggestion
    auto used = m_parser.AddData(pData, iSize, &m_buffer, &m_dataSize);
```

fuzzard replied:
there was a discussion around this sort of change not too long ago. I think i am on the side of ints a basic type, why go auto when its longer than int to type

kambala-decapitator reviewed (commented):
isn't it possible to replace `= 0` with `{}`?

fuzzard commented:
Any complaints with this as is? 
If not, i'd like to merge it this weekend

howie-f commented on xbmc/dialogs/GUIDialogMediaFilter.cpp:
```diff
@@ -474,7 +474,9 @@ void CGUIDialogMediaFilter::InitializeSettings()
       if (filter.settingType == SettingType::Integer)
       {
-        int min, interval, max;
+        int min = 0;
```
no complaints, just a remark https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md#54-initialization => `int min{};`
would be in line with the arrays then, but honestly my personal preference is your version
